thunder/examine/__init__.py

```diff
@@ -216,3 +216,48 @@ def get_fusion_symbols(trace: TraceCtx, warn_if_fusions_unavailable: bool = True
             fusions.append(bsym)
 
     return fusions
+
+
+def get_nvFuser_repro(trace: TraceCtx, fusion_name: str, /) -> str:
+    """
+    Helper function to get the repro of a specific nvFusion segment.
+    """
+    if (fusion := trace.python_ctx().get(fusion_name, None)) is None:
+        available_fusions = [name for name, _ in get_fusions(trace)]
+        assert False, f"Unable to find fusion '{fusion_name}' in trace. Available fusions are: {available_fusions}."
+
+    if fusion.last_used is None:
+        assert (
+            False
+        ), "Fusion definition needs to be executed to record the inputs. You must execute the trace first before querying the repro."
+
+    msg = ""
+    msg += (
+        "import torch\nfrom nvfuser import FusionDefinition, DataType\n"
+        f"{fusion.last_used}"
+        "with FusionDefinition() as fd:\n"
+        f"    nvfuser_fusion_id{fusion.last_used.id()}(fd)\n\n"
+    )
+
+    msg += "inputs = [\n"
+    for size, stride, dtype, device in fusion.last_inputs_meta:
```

Comment:
I'm fine with copying these bits from nvFuser, but I wonder if we should use it from nvFuser eventually.